src/main/java/EchoServer.java

```diff
@@ -0,0 +1,29 @@
+import echoserver.ClientConnection;
+import echoserver.Console;
+import echoserver.HostServer;
+import echoserver.Presenter;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+
+public class EchoServer {
+    public static void main(String[] args) throws IOException {
+        Presenter hostPresenter = new Presenter(new Console(System.out));
+        ServerSocket ss = new ServerSocket(100);
+        HostServer hs = new HostServer(ss);
+
+        hostPresenter.showAwaitingInputMessage();
+        ClientConnection connectedClient = hs.listenForConnection();
+        connectedClient.promptInput();
+
+        hostPresenter.displayInputFromSocket(connectedClient);
+
+        Console clientConsole = new Console(connectedClient.getOutputStream());
+        Presenter clientPresenter = new Presenter(clientConsole);
+        clientPresenter.displayInputFromSocket(connectedClient);
+
+        clientPresenter.showConnectionClosingMessage();
+        hs.close();
+    }
```

Comment:
There is quite a bit of procedure here. I would have expected an API that communicates the domain a little more.

```java
// Setup
new EchoServer(connection).echo();
// Teardown
```

Or something of that nature.

Reply:
Or perhaps something like this would make more sense.

```java
public static void main(String [] args) {
    // Setup
    new EchoServer(hostServer).serve();
    // Tear down
}
```

Reply:
Kind of along the same lines as Damon's comment, it's good to keep your `main()` methods as slim as possible. I would suggest extracting as much behavior out of `main()` into components that can be unit-tested.